.github: Migrate XLA tests to GHA

seemethere opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #64320
* #68817

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D30684490](https://our.internmc.facebook.com/intern/diff/D30684490)

cc @ezyang @seemethere @malfet @walterddr @lg20987 @pytorch/pytorch-dev-infra @bdhirsh

seemethere committed:
.github: Migrate XLA tests to GHA

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

[ghstack-poisoned]

seemethere commented:
@seemethere has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30684490).

seemethere committed:
Update on ".github: Migrate XLA tests to GHA"


Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D30684490](https://our.internmc.facebook.com/intern/diff/D30684490)

cc @ezyang @seemethere @malfet @walterddr @lg20987 @pytorch/pytorch-dev-infra @bdhirsh

[ghstack-poisoned]

seemethere commented:
@seemethere has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30684490).

seemethere commented:
This PR is blocked by https://github.com/pytorch/xla/issues/3107

seemethere committed:
Update on ".github: Migrate XLA tests to GHA"


Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D30684490](https://our.internmc.facebook.com/intern/diff/D30684490)

cc @ezyang @seemethere @malfet @walterddr @lg20987 @pytorch/pytorch-dev-infra @bdhirsh

[ghstack-poisoned]

seemethere commented:
@seemethere has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30684490).

seemethere committed:
Update on ".github: Migrate XLA tests to GHA"


Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D30684490](https://our.internmc.facebook.com/intern/diff/D30684490)

cc @ezyang @seemethere @malfet @walterddr @lg20987 @pytorch/pytorch-dev-infra @bdhirsh

[ghstack-poisoned]

seemethere committed:
Update on ".github: Migrate XLA tests to GHA"


Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D30684490](https://our.internmc.facebook.com/intern/diff/D30684490)

cc @ezyang @seemethere @malfet @walterddr @lg20987 @pytorch/pytorch-dev-infra @bdhirsh

[ghstack-poisoned]

seemethere committed:
Update on ".github: Migrate XLA tests to GHA"


Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D30684490](https://our.internmc.facebook.com/intern/diff/D30684490)

cc @ezyang @seemethere @malfet @walterddr @lg20987 @pytorch/pytorch-dev-infra @bdhirsh

[ghstack-poisoned]

seemethere committed:
Update on ".github: Migrate XLA tests to GHA"


Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D30684490](https://our.internmc.facebook.com/intern/diff/D30684490)

cc @ezyang @seemethere @malfet @walterddr @lg20987 @pytorch/pytorch-dev-infra @bdhirsh

[ghstack-poisoned]

seemethere committed:
Update on ".github: Migrate XLA tests to GHA"


Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D30684490](https://our.internmc.facebook.com/intern/diff/D30684490)

cc @ezyang @seemethere @malfet @walterddr @lg20987 @pytorch/pytorch-dev-infra @bdhirsh

[ghstack-poisoned]

seemethere commented:
@seemethere has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30684490).

seemethere committed:
Update on ".github: Migrate XLA tests to GHA"


Signed-off-by: Eli Uriegas <eliuriegasfb.com>

Differential Revision: [D30684490](https://our.internmc.facebook.com/intern/diff/D30684490)

cc ezyang seemethere malfet walterddr lg20987 pytorch/pytorch-dev-infra bdhirsh

[ghstack-poisoned]

seemethere commented:
@seemethere has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30684490).

seemethere committed:
Update on ".github: Migrate XLA tests to GHA"


Signed-off-by: Eli Uriegas <eliuriegasfb.com>

Differential Revision: [D30684490](https://our.internmc.facebook.com/intern/diff/D30684490)

cc ezyang seemethere malfet walterddr lg20987 pytorch/pytorch-dev-infra bdhirsh

[ghstack-poisoned]

seemethere committed:
Update on ".github: Migrate XLA tests to GHA"


Signed-off-by: Eli Uriegas <eliuriegasfb.com>

Differential Revision: [D30684490](https://our.internmc.facebook.com/intern/diff/D30684490)

cc ezyang seemethere malfet walterddr lg20987 pytorch/pytorch-dev-infra bdhirsh

[ghstack-poisoned]

seemethere committed:
Update on ".github: Migrate XLA tests to GHA"


Signed-off-by: Eli Uriegas <eliuriegasfb.com>

Differential Revision: [D30684490](https://our.internmc.facebook.com/intern/diff/D30684490)

cc ezyang seemethere malfet walterddr lg20987 pytorch/pytorch-dev-infra bdhirsh

[ghstack-poisoned]

seemethere committed:
Update on ".github: Migrate XLA tests to GHA"


Signed-off-by: Eli Uriegas <eliuriegasfb.com>

Differential Revision: [D30684490](https://our.internmc.facebook.com/intern/diff/D30684490)

cc ezyang seemethere malfet walterddr lg20987 pytorch/pytorch-dev-infra bdhirsh

[ghstack-poisoned]

seemethere committed:
Update on ".github: Migrate XLA tests to GHA"


Signed-off-by: Eli Uriegas <eliuriegasfb.com>

Differential Revision: [D30684490](https://our.internmc.facebook.com/intern/diff/D30684490)

cc ezyang seemethere malfet walterddr lg20987 pytorch/pytorch-dev-infra bdhirsh

[ghstack-poisoned]

seemethere committed:
Update on ".github: Migrate XLA tests to GHA"


Signed-off-by: Eli Uriegas <eliuriegasfb.com>

Differential Revision: [D30684490](https://our.internmc.facebook.com/intern/diff/D30684490)

cc ezyang seemethere malfet walterddr lg20987 pytorch/pytorch-dev-infra bdhirsh

[ghstack-poisoned]

seemethere committed:
Update on ".github: Migrate XLA tests to GHA"


Signed-off-by: Eli Uriegas <eliuriegasfb.com>

Differential Revision: [D30684490](https://our.internmc.facebook.com/intern/diff/D30684490)

cc ezyang seemethere malfet walterddr lg20987 pytorch/pytorch-dev-infra bdhirsh

[ghstack-poisoned]

seemethere commented:
@seemethere has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30684490).

seemethere committed:
Update on ".github: Migrate XLA tests to GHA"


Signed-off-by: Eli Uriegas <eliuriegasfb.com>

Differential Revision: [D30684490](https://our.internmc.facebook.com/intern/diff/D30684490)

cc ezyang seemethere malfet walterddr lg20987 pytorch/pytorch-dev-infra bdhirsh

[ghstack-poisoned]

seemethere committed:
Update on ".github: Migrate XLA tests to GHA"


Signed-off-by: Eli Uriegas <eliuriegasfb.com>

Differential Revision: [D30684490](https://our.internmc.facebook.com/intern/diff/D30684490)

cc ezyang seemethere malfet walterddr lg20987 pytorch/pytorch-dev-infra bdhirsh

[ghstack-poisoned]

seemethere committed:
Update on ".github: Migrate XLA tests to GHA"


Signed-off-by: Eli Uriegas <eliuriegasfb.com>

Differential Revision: [D30684490](https://our.internmc.facebook.com/intern/diff/D30684490)

cc ezyang seemethere malfet walterddr lg20987 pytorch/pytorch-dev-infra bdhirsh

[ghstack-poisoned]

seemethere committed:
Update on ".github: Migrate XLA tests to GHA"


Signed-off-by: Eli Uriegas <eliuriegasfb.com>

Differential Revision: [D30684490](https://our.internmc.facebook.com/intern/diff/D30684490)

cc ezyang seemethere malfet walterddr lg20987 pytorch/pytorch-dev-infra bdhirsh

[ghstack-poisoned]

seemethere commented on .github/templates/linux_ci_workflow.yml.j2:
```diff
@@ -240,10 +240,15 @@ jobs:
             TEST_COMMAND=.jenkins/pytorch/test.sh
           fi
+          PROXY_ENV=
+          if [[ $TEST_CONFIG != 'xla' ]]; then
```
Squid proxy behaves badly for xla multiprocessing, see https://github.com/pytorch/xla/issues/3107, disabling squid proxy here for that reason

malfet replied:
Can you please add it as the comment to the template?

seemethere commented:
@seemethere has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30684490).

seemethere committed:
Update on ".github: Migrate XLA tests to GHA"


Signed-off-by: Eli Uriegas <eliuriegasfb.com>

Differential Revision: [D30684490](https://our.internmc.facebook.com/intern/diff/D30684490)

cc ezyang seemethere malfet walterddr lg20987 pytorch/pytorch-dev-infra bdhirsh

[ghstack-poisoned]

seemethere commented:
@seemethere has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30684490).

atalman commented on .jenkins/pytorch/test.sh:
```diff
@@ -518,8 +533,9 @@ fi
+elif [[ "${TEST_CONFIG}" == *xla* ]]; then
   install_torchvision
+  build_xla
   test_xla
```
Since we are running test_xla right after build_xla . Why do we need to clone_pytorch_xla() again inside test_xla ? Also which method is now running the tests in test_xla(), sorry its not clear.

seemethere replied:
It's actually a no-op since it checks for the directories existence

seemethere replied:
`test_xla` is still a function within `.jenkins/pytorch/test.sh` it doesn't show up in this diff since it was mostly unmodified

atalman replied:
Got ti Thanks!

seemethere committed:
Update on ".github: Migrate XLA tests to GHA"


Signed-off-by: Eli Uriegas <eliuriegasfb.com>

Differential Revision: [D30684490](https://our.internmc.facebook.com/intern/diff/D30684490)

cc ezyang seemethere malfet walterddr lg20987 pytorch/pytorch-dev-infra bdhirsh

[ghstack-poisoned]

seemethere commented:
@seemethere has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30684490).

seemethere committed:
Update on ".github: Migrate XLA tests to GHA"


Signed-off-by: Eli Uriegas <eliuriegasfb.com>

Differential Revision: [D30684490](https://our.internmc.facebook.com/intern/diff/D30684490)

cc ezyang seemethere malfet walterddr lg20987 pytorch/pytorch-dev-infra bdhirsh

[ghstack-poisoned]

seemethere commented:
@seemethere has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30684490).
